lib/sqlalchemy/ext/automap.py

```diff
@@ -1031,16 +1143,21 @@ def _is_many_to_many(automap_base, table):
 
 
 def _relationships_for_fks(
-    automap_base,
-    map_config,
-    table_to_map_config,
-    collection_class,
-    name_for_scalar_relationship,
-    name_for_collection_relationship,
-    generate_relationship,
-):
-    local_table = map_config.local_table
-    local_cls = map_config.cls  # derived from a weakref, may be None
+    automap_base: Type[Any],
+    map_config: _DeferredMapperConfig,
+    table_to_map_config: Union[
+        Dict[Optional[Table], _DeferredMapperConfig],
+        Dict[Table, _DeferredMapperConfig],
+    ],
+    collection_class: type,
+    name_for_scalar_relationship: NameForScalarRelationshipType,
+    name_for_collection_relationship: NameForCollectionRelationshipType,
+    generate_relationship: GenerateRelationshipType,
+) -> None:
+    local_table = cast(Union[Table, None], map_config.local_table)
```

Comment:
**mike bayer** (zzzeek) wrote:

for casts, if the type is a generic type (e.g. has [] brackets), we put the type into quotes.  This is because the type inside of `cast()` is invoked at runtime, and the construction of a generic type like `Union[Type, None]` has been shown to be relatively slow.   

Also for `Union[Table, None]`, we use `Optional[Table]`.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4271